packages/odie-client/src/components/message/sources.tsx

```diff
@@ -29,7 +29,7 @@ export const Sources = ( { message }: { message: Message } ) => {
 			];
 		}
 		return [];
-	}, [ message?.context?.sources ] );
+	}, [ message?.context?.sources, trackEvent ] );
```

Comment:
Why is this necessary?

Reply:
That's a dependencies array, inside a `useMemo` method.

`useMemo` is meant to memoize calculations, so for this case, we process the data once and we store it as `const sources;` If sources changes, we update the const for the next render. if trackEvent function changes, we need to re "memoize" the calculations. 

This won't happen (we change that trackEvent function through the lifetime of the chat), but the correct thing is to always include all the dependencies in the array. I realized it was t throwing a warning so I fixed it.

![image](https://github.com/user-attachments/assets/135695d2-479d-491c-b532-fa53ba55b1aa)

Reply:
So, in the really unlikely chance of changing the `trackEvent` function through the chat, all the events will be sent through the new function :)

Reply:
Thanks for the explanation! That is very helpful.